src/components/Tabs/TabsNav.test.tsx

```diff
@@ -27,30 +28,101 @@ afterEach(() => {
-  const links = getAllByTestId('link');
-  expect(links.length).toEqual(MOCK_TABS.length);
-  links.forEach(l => expect(l.classList).toContain('Tab'));
-});
+      expect(tabNavContainer.length).toBe(1);
+    });
 
-test('should apply Selected class name on selected Link', () => {
-  const { container } = renderComponent();
-  const selected = container.getElementsByClassName('Selected');
-  expect(selected.length).toBe(1);
-  expect(selected[0].textContent).toBe(MOCK_TABS[0].text);
-});
+    test('should apply List class only to the List element', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} />);
+      const list = container.getElementsByClassName(styles.List);
 
-test('should apply replace prop to Link', () => {
-  renderComponent({ tabs: [MOCK_TABS[0]] });
-  expect(mockLink).toHaveBeenCalledTimes(1);
-  expect(mockLink).toHaveBeenCalledWith(expect.objectContaining({ replace: true }));
-});
+      expect(list.length).toBe(1);
+    });
+
+    test('should apply Tab class only to the Link elements', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} />);
+      const link = container.getElementsByClassName(styles.Tab);
+      expect(link.length).toBe(3);
+    });
+
+    test('should apply Selected class on default selected link', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} />);
+      const selectedLink = container.getElementsByClassName(styles.Selected);
+
+      expect(selectedLink[0]).toHaveTextContent(MOCK_TABS[0].text);
+    });
+
+    test('should apply Selected class only to the selected link', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} />);
+      const selectedLink = container.getElementsByClassName(styles.Selected);
+
+      expect(selectedLink.length).toBe(1);
+      expect(selectedLink[0]).not.toHaveTextContent(MOCK_TABS[1].text);
+      expect(selectedLink[0]).not.toHaveTextContent(MOCK_TABS[2].text);
+    });
+
+    test('should not apply Selected class on any Link by default', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} location={undefined} />);
+      const selected = container.getElementsByClassName(styles.Selected);
+
+      expect(selected.length).toBe(0);
+    });
+  });
+
+  describe('list', () => {
+    test('should render correct number of links', () => {
+      const { container } = render(<TabsNav {...DEFAULT_PROPS} />);
+      const tab = container.getElementsByClassName(styles.Tab);
+
+      expect(tab.length).toEqual(MOCK_TABS.length);
+    });
+
+    test('should render links in the correct order', () => {
+      const MOCK_TABS: TabNav[] = [
+        { text: 'Tab 1,', to: '/1' },
+        { text: 'Tab 2,', to: '/2' },
+        { text: 'Tab 3', to: '/3' }
+      ];
+
+      const { container } = render(<TabsNav tabs={MOCK_TABS} location={MOCK_TABS[0].to} />);
+
+      expect(container.textContent).toBe('Tab 1,Tab 2,Tab 3');
+    });
+  });
+
+  describe('link/tab', () => {
```

Comment:
Can we change this block to `react-router-dom links` or similar? Might be more descriptive.

Reply:
Indeed! Updated.